File: 804-unique-morse-code-words/804-unique-morse-code-words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        
        vector<string> morse ={".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.." } ; 
        unordered_set<string> box ; 
        for(auto word:words){
            string code ;
            for(auto w:word){
                code += morse[w-'a'] ;}
            box.insert(code) ; 
            }
        return box.size() ; 
        }
        
    
};

File: 804-unique-morse-code-words/804-unique-morse-code-words.py
class Solution:
    def uniqueMorseRepresentations(self, words: List[str]) -> int:
        cv = {} 
        morse = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        for i in range(97,123):
            cv[chr(i)] = morse[i-97]
        
        box = set() 
        for word in words: 
            t = ''
            for w in word: 
                t += cv[w]
            box.add(t)
        
        return len(box)